### DISCUSSION TITLE
현재 CICD 과정에 대해서 피드백을 얻고 싶습니다.

### DISCUSSION THREAD

**Discussion by 02ggang9** [BDD-CLUB/01-doo-re-back#51] [논의 사항]
현재 GitHub Action에서 돌아가는 코드는 CI/CD가 동시에 진행되고 있습니다. 과거 KEEPER가 이러한 방식을 사용했었는데 아래와 같은 불편함을 느꼈습니다.

코드 리뷰를 다 받고, 머지 버튼을 누르면 자동으로 CICD가 돌아가는데 테스트에 실패했을 경우 PR이 Close되고 머지는 되지 않았습니다. 이럴 경우에 또 다시 PR을 올리고 머지 버튼을 누르면서 확인할 수 밖에 없었습니다. 연속으로 2번 실패하면 대역죄인이 된 기분이었습니다..ㅠ

두레 CICD는 로컬에서 테스트를 마치고 머지 버튼을 눌러 머지한 후 CICD를 수동으로 돌려 develop 브랜츠 코드의 테스트를 진행하고 자동 배포를 진행하고 있습니다. 이렇게 하면 발생하는 문제점이 로컬에서 테스트가 통과해 문제가 없다고 생각했지만 서버에서는 문제가 발생할 수 있는 코드가 머지됩니다. 

CICD 코드를 CI, CD 코드로 분리해 머지하기 전 자신의 브랜치를 CI 코드를 통해 이상이 없는지 확인하고, 안정화 된 코드(Develop 브랜치 코드)를 CD해 자동배포 하는 방법은 어떻게 생각하시는지 궁금합니다!

제가 생각한 방법의 단점은 까먹고 CI 코드를 통해 확인하지 못 하고 머지할 수 있다는 점입니다. 하지만 CI/CD 코드를 분리한다면 이런 경우를 대비한 코드 작성은 쉽다고 생각이 듭니다!

더 나은 두레의 CICD를 위해 피드백을 남겨주시면 정말로 감사드리겠습니다.

**Comment by JJimini** [comment]
CI와 CD를 분리하되, CI는 자동(pr올릴 시 자동 + commit 새로 올라와도 자동) CD는 수동으로 한다.
